File: solutions/442.Find_All_Duplicates_in_an_Array/442.Find_All_Duplicates_in_an_Array.ts
function findDuplicates(nums: number[]): number[] {
  const result:number[] = []
  for (let i = 0; i < nums.length; i++) {
      let index = Math.abs(nums[i]) - 1
      let value = nums[index]
      if(value < 0) {
          result.push(Math.abs(nums[i]))
      } else {
          nums[index] = -value
      }
  }

  return result
};

File: solutions/medium/442.Find_All_Duplicates_in_an_Array/442.Find_All_Duplicates_in_an_Array.py
from ast import List


class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        answer = []

        for i in range(len(nums)):
            index = abs(nums[i]) - 1
            value = nums[index]
            if value < 0:
                answer.append(abs(value))
            else:
                nums[index] = -nums[index]

        return answer
